src/fate.ts

```diff
@@ -212,7 +212,6 @@ export class Fate<T = unknown> {
 	 */
 	public setErrorCode(code: string): Fate<T> {
 		this.errorCode(code);
-		this.success(false);
 		this.done(true);
 
 		return this;
```

Comment:
Is `return this` called twice here, or is my editor showing a duplicate? If `return this` appears twice we can remove one as the second and everything below it will be unreachable code.